lib/sdf-server/src/server/service/component/debug.rs

```diff
@@ -57,55 +39,58 @@ pub struct DebugComponentRequest {
 
-    // let debug_view = ComponentDebugView::new(&ctx, &component).await?;
+    let debug_view = ComponentDebugView::new(&ctx, &component).await?;
 
-    let attributes = vec![];
+    let mut attributes = vec![];
     let input_sockets = vec![];
     let output_sockets = vec![];
 
-    // let transform_start = Instant::now();
-    //
-    // for attribute_debug in debug_view.attributes {
-    //     attributes.push(FrontendAttributeDebugView {
-    //         name: attribute_debug
-    //             .prop
-    //             .as_ref()
-    //             .map(|p| p.name())
-    //             .unwrap_or("")
-    //             .into(),
-    //         path: attribute_debug.path.to_owned(),
-    //         debug_data: get_attribute_metadata(&ctx, attribute_debug).await?,
-    //     });
-    // }
-    //
-    // for attribute_debug in debug_view.input_sockets {
-    //     input_sockets.push(FrontendAttributeDebugView {
-    //         name: attribute_debug.path.to_owned(),
-    //         path: "Input Socket".into(),
-    //         debug_data: get_attribute_metadata(&ctx, attribute_debug).await?,
-    //     });
-    // }
-    //
-    // for attribute_debug in debug_view.output_sockets {
-    //     output_sockets.push(FrontendAttributeDebugView {
-    //         name: attribute_debug.path.to_owned(),
-    //         path: "Output Socket".into(),
-    //         debug_data: get_attribute_metadata(&ctx, attribute_debug).await?,
-    //     });
-    // }
-    //
-    // dbg!(transform_start.elapsed());
+    let transform_start = Instant::now();
+    let mut cache: Vec<AttributeValueId> = vec![];
+    //grab attribute value debug views
+    for (av, children) in debug_view.attribute_tree {
+        if !cache.contains(&av) {
+            let avd = AttributeDebugView::new(&ctx, av, None, None).await?;
+            if let Some(name) = avd.prop.as_ref().map(|p| p.name.to_owned()) {
+                attributes.push(FrontendAttributeDebugView {
+                    name,
+                    path: avd.path.to_owned(),
+                    debug_data: avd,
+                });
+                cache.push(av);
+            }
+        }
+
+        for child_av in children {
+            if !cache.contains(&child_av) {
+                let child_avd = AttributeDebugView::new(&ctx, child_av, None, Some(av)).await?;
+                if let Some(name) = child_avd.prop.as_ref().map(|p| p.name.to_owned()) {
+                    attributes.push(FrontendAttributeDebugView {
+                        name,
+                        path: child_avd.path.to_owned(),
+                        debug_data: child_avd,
+                    });
+                    cache.push(child_av);
+                }
+            }
+        }
+    }
+    attributes.sort_by_key(|view| view.path.to_lowercase());
+
+    //grab input socket info next
+    //grab output socket info next
+
+    dbg!(transform_start.elapsed());
```

Comment:
Let's delete these or put this behind an `info` or `debug` statement with context for what it's benchmarking.